test/e2e_node/conformance_test.go

```diff
@@ -99,66 +97,85 @@ var _ = Describe("Container Conformance Test", func() {
 			})
 		})
 		Context("when running a container that terminates", func() {
-			var terminateCase ConformanceContainer
-			It("it should run successfully to completion [Conformance]", func() {
-				terminateCase = ConformanceContainer{
-					Container: api.Container{
-						Image:           "gcr.io/google_containers/busybox",
-						Name:            "busybox",
-						Command:         []string{"sh", "-c", "env"},
-						ImagePullPolicy: api.PullIfNotPresent,
-					},
-					Client:   cl,
-					Phase:    api.PodSucceeded,
-					NodeName: *nodeName,
-				}
-				err := terminateCase.Create()
+			It("should be able to create, inspect and delete it [Conformance]", func() {
+				c := NewConformanceContainer(cl, *nodeName, api.Container{
+					Image:           "gcr.io/google_containers/busybox:1.24",
+					Name:            "terminate-container",
+					Command:         []string{"sh", "-c", "env"},
+					ImagePullPolicy: api.PullIfNotPresent,
+				})
+
+				By("create the container")
+				defer c.Delete()
+				err := c.Create()
 				Expect(err).NotTo(HaveOccurred())
 
+				By("wait up to 2m for the container to become  terminated")
 				// TODO: Check that the container enters running state by sleeping in the container #23309
-				Eventually(func() (api.PodPhase, error) {
-					pod, err := terminateCase.Get()
-					return pod.Phase, err
-				}, retryTimeout, pollInterval).Should(Equal(terminateCase.Phase))
-			})
-			It("it should report its phase as 'succeeded' [Conformance]", func() {
-				ccontainer, err := terminateCase.Get()
+				c.Wait(2*time.Minute, func(s *api.ContainerStatus) bool { return s.State.Terminated != nil })
+
+				By("check the container status")
+				status, err := c.Status()
+				Expect(err).NotTo(HaveOccurred())
+				Expect(isContainerSucceed(status)).Should(BeTrue())
+
+				By("delete the container")
+				err = c.Delete()
 				Expect(err).NotTo(HaveOccurred())
-				Expect(ccontainer).Should(CContainerEqual(terminateCase))
 			})
-			It("it should be possible to delete [Conformance]", func() {
-				err := terminateCase.Delete()
+			It("should report termination message if TerminationMessagePath is set [Conformance]", func() {
+				terminationMessage := "DONE"
+				terminationMessagePath := "/dev/termination-log"
+				c := NewConformanceContainer(cl, *nodeName, api.Container{
+					Image:   "gcr.io/google_containers/busybox:1.24",
+					Name:    "termination-message-container",
+					Command: []string{"/bin/sh", "-c"},
+					Args:    []string{"sleep 60 && /bin/echo -n " + terminationMessage + " > " + terminationMessagePath},
+					TerminationMessagePath: terminationMessagePath,
```

Comment:
Do you need to sleep 60 seconds?  This is going to extend the runtime of the tests by a good bit.